playground/backend/pkg/executors/java_executor.go

```diff
@@ -0,0 +1,64 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package executors
+package executors
+
+import (
+	"beam.apache.org/playground/backend/pkg/api/pathchecker"
+	"beam.apache.org/playground/backend/pkg/fs_tool"
+	"os/exec"
+	"path/filepath"
+	"strings"
+)
+
+const (
+	beamJarPath   = "/opt/apache/beam/jars/beam-sdks-java-harness.jar"
+	runnerJarPath = "/opt/apache/beam/jars/beam-runners-direct.jar"
+	slf4jPath     = "/opt/apache/beam/jars/slf4j-jdk14.jar"
+)
+
+// JavaExecutor for Java code
+type JavaExecutor struct {
+	Lang string
+	fs   fs_tool.JavaFileSystemService
+}
+
+// Validate checks that the file exists and that extension of the file matches the SDK
+func (javaExec JavaExecutor) Validate(fileName string) (bool, error) {
+	filePath := filepath.Join(javaExec.fs.GetJavaSrcPath(), fileName)
+	return pathchecker.CheckPathIsValid(filePath, javaExec.Lang)
+}
+
+// Compile makes compilation of the java code
+func (javaExec JavaExecutor) Compile(fileName string) error {
+	cmd := exec.Command("javac", "-d", javaExec.fs.GetJavaBinPath(), "-classpath", beamJarPath,
+		filepath.Join(javaExec.fs.GetJavaSrcPath(), fileName))
+	_, err := cmd.Output()
+	return err
+}
+
+// Run executes the compiled java code
+func (javaExec JavaExecutor) Run(className string) (string, error) {
+	fullClassPath := strings.Join([]string{javaExec.fs.GetJavaBinPath(), beamJarPath, runnerJarPath, slf4jPath}, ":")
+	cmd := exec.Command("java", "-cp", fullClassPath, className)
+	out, err := cmd.Output()
+	return string(out), err
+}
+
+// NewJavaExecutor creates new instance of java executor
+func NewJavaExecutor(fs *fs_tool.JavaFileSystemService) *JavaExecutor {
+	return &JavaExecutor{".java", *fs}
```

Comment:
If the first argument of `JavaExecutor` is always ".java" will it be better to use constant and remove `Lang` parameter from `JavaExecutor?